include/mongo_protocol.hrl

```diff
@@ -44,6 +45,25 @@
   projector = #{} :: mc_worker_api:projector()
 }).
 
+-record(op_msg_write_op, {
+  command :: command(),
+  collection :: colldb(),
+  database :: [] | mc_worker_api:database(),
+  extra_fields = [] :: bson:document() | nonempty_list({binary(),any()}),
+  documents_name = <<"documents">> :: bson:utf8(),
+  documents = [] :: any()
+}).
+
+-record(op_msg_response, {
+  response_doc :: map()
+}).
+
+-record(op_msg_command, {
+  database :: [] | mc_worker_api:database(),
```

Comment:
```suggestion
  database = [] :: mc_worker_api:database(),
```

Reply:
I made undefined part of the type. See commit below.